src/server/routes/contract/extensions/erc1155/write/claimTo.ts

```diff
@@ -86,12 +87,19 @@ export async function erc1155claimTo(fastify: FastifyInstance) {
         tokenId: BigInt(tokenId),
       });
 
+      let data: Hex;
+      try {
+        data = await encode(transaction);
+      } catch (e) {
+        throw createCustomError(`${e}`, StatusCodes.BAD_REQUEST, "BAD_REQUEST");
+      }
+
       let queueId: string;
       const insertedTransaction = {
         chainId,
         from: fromAddress as Address,
         to: contractAddress as Address | undefined,
-        data: (await resolvePromisedValue(transaction.data)) as Hex,
+        data,
```

Comment:
We're you doing this pattern in other places too? You should always use encode to get the calldata from a prepared tx 🙏

Reply:
👍 There's two more places we're using `resolvePromisedValue`. I'll look to replace it with `encode`. Didn't know about encode!